src/v/cloud_storage/cache_service.cc

```diff
@@ -107,11 +109,13 @@ ss::future<> cache::consume_cache_space(size_t sz) {
     vassert(ss::this_shard_id() == 0, "This method can only run on shard 0");
     _current_cache_size += sz;
     if (_current_cache_size > _max_cache_size) {
-        auto units = ss::try_get_units(_cleanup_sm, 1);
-        if (units) {
-            co_await clean_up_cache();
+        if (ss::lowres_clock::now() - _last_clean_up > min_clean_up_interval) {
```

Comment:
one thing to watch out for here which i recently learned is that lowres_clock has std::steady_clock semantics, so it's value could be close to 0 right after bootup, making statements like this more sensitive to how _last_clean_up is initialized.

i think this case looks ok but i got bit by something like this recently.